lib/libc/minimal/source/stdlib/rand.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2021 Space Cubics, LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <toolchain.h>
+
+#define OUTPUT_BITS (0x7fffffffU)
+#define MULTIPLIER (1103515245U)
+#define INCREMENT (12345U)
+
+static uint32_t __seed;
+
+void _srand(unsigned int s)
+{
+	/* seed of 0 doesn't generate a good random number */
+	if (s == 0) {
+		s = 1;
+	}
+	__seed = s;
+}
+__weak FUNC_ALIAS(_srand, srand, void);
```

Comment:
What's the reason for defining underscore version of a function, and then add a weak alias with normal name? (Why not use the normal name directly?)

Reply:
I followed other functions in minimal libc.  Why do they have aliases?

Reply:
> I followed other functions in minimal libc. Why do they have aliases?

Where do you see that?

```
zephyr/lib/libc/minimal/source$ grep -r -i weak *
zephyr/lib/libc/minimal/source$ grep -r -i alias *
```

That's exactly my point - existing sources in zephyr/lib/libc/minimal/source/ **don't** have any aliasing like that, and I'd suggest to not add it here either.

Reply:
Oops! Thanks.  I was checking `lib/libc/newlib/libc-hooks.c`.  Fixed.

Reply:
> Oops! Thanks. I was checking lib/libc/newlib/libc-hooks.c. Fixed.

I see. That's actually for newlib libc. Thanks for update!